### DISCUSSION TITLE
How configuring PnP Search Box to lookup inside metadata field of documents

### DISCUSSION THREAD

**Discussion by apolorc** [microsoft-search/pnp-modern-search#3426] [Q&A]
Hi guys!

We have two Sharepoint online sites:
- A site is a container of document libraries, configured with metadata fields, created as site columns and associated with RefinablesString. In the documentary libraries, indexes have been created for each of the metadata fields. Some metadata fileds are lookup type, others text.
- The other site contains search pages where different tools offered by Pnp Modern Search are being used. The pages are configured correctly with their result boxes and filters, but the search box only searches on the title of the elements and on the contents of the documents and not on the content of the metadata fields. The PnP Search Result is associated with the search box.

For example, we have a "code" metadata field.  "Code" is a text with an identifier for each document; We can filter the values of that field with the PnP Filter, but if a code value is placed in the PnP Search Box it does not return any results in result search box

How could the PnP Search Box be configured to also search within the metadata content?

Can you please help me?

**Comment by kasperbolarsen** [comment]
Let me see if I understand your question: let's assume that the code managed property is mapped to RefinableString01, and one of the code values is "ISO14001".
If you enter RefinableString01:ISO14001 in the search box you get the results you are looking for?

**Comment by apolorc** [comment]
Hi,
Thanks for your answer. 

Yes, if I enter RefinableStringXXX:codevalue in the search box, I get results; but I need do not use "RefinableStringXXX" in pathern to search.

**Comment by kasperbolarsen** [comment]
The issue is that the RefinableString property isn't searchable and you don't have an autogenerated managed property for that field. 
The solution is to create a new managed property in the SharePoint Admin center:
![image](https://github.com/microsoft-search/pnp-modern-search/assets/20593570/4f535489-7919-4f04-aafc-9912049103f0)

In this case I have a single value Managed metadata field "search_TaxonomySingle" which have created the crawled property SEARCH_TAXONOMYSINGLE which I have mapped to a RefinableString for filtering, but as I also want to search for that metadata I have added a new Manage Property SeachTaxonomySingle and mapped it to the same crawled property.
Once SeachTaxonomySingle contains values you should add it to "Selected Properties" in the Results web part and you will now be able to search for the values in that field :-)

**Comment by apolorc** [comment]
Hi @kasperbolarsen,

Thanks a lot for your answer. 

I am going to implement you suggestion
